src/sv_primlit_integral.rs

```diff
@@ -138,6 +153,34 @@ impl SvPrimaryLiteralIntegral {
         ret
     }
 
+    /* Returns whether the MSB of data_01 is high. The size must be correctly specified. */
+    pub fn data_01_msb_high(&self) -> bool {
+        let left_leading_zeros: usize =
+            usize::BITS as usize - (self.size - (self.data_01.len() - 1) * usize::BITS as usize);
+
+        if self.data_01[0].leading_zeros() as usize == left_leading_zeros {
+            true
+        } else {
+            false
+        }
+    }
+
+    /* Returns whether the MSB of data_xz is high. The size must be correctly specified. */
+    pub fn data_xz_msb_high(&self) -> bool {
+        if !self.is_4state() {
+            return false;
+        }
```

Comment:
Early return could/should be replaced by something like:
```rust
if self.is_4state() {
    let left_leading_zeros = ...;
    if (...) {
        true
    } else {
        false
    }
} else {
    false
}
```
Avoiding early returns (using the `return` keyword) simplifies control flow analysis.

Reply:
Fixing that as well.